kroxylicious-sample/src/test/java/io/kroxylicious/sample/SampleFilterIntegrationTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.sample;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.kroxylicious.proxy.config.FilterDefinitionBuilder;
+import io.kroxylicious.test.tester.KroxyliciousTester;
+import io.kroxylicious.testing.kafka.api.KafkaCluster;
+import io.kroxylicious.testing.kafka.junit5ext.KafkaClusterExtension;
+
+import static io.kroxylicious.test.tester.KroxyliciousConfigUtils.proxy;
+import static io.kroxylicious.test.tester.KroxyliciousConfigUtils.withDefaultFilters;
+import static io.kroxylicious.test.tester.KroxyliciousTesters.kroxyliciousTester;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@ExtendWith(KafkaClusterExtension.class)
+public class SampleFilterIntegrationTest {
+
+    private static final String PRE_TRANSFORM_VALUE = "foo";
+    private static final String NO_TRANSFORM_VALUE = "sample";
+    private static final String PRODUCE_TRANSFORM_VALUE = "bar";
+    private static final String FETCH_TRANSFORM_VALUE = "baz";
+    private static final String FIND_CONFIG_FIELD = "findValue";
+    private static final String REPLACE_CONFIG_FIELD = "replaceValue";
+    private static final Map<String, Object> PRODUCE_CONFIG = Map.of(FIND_CONFIG_FIELD, "foo", REPLACE_CONFIG_FIELD, "bar");
+    private static final Map<String, Object> FETCH_CONFIG = Map.of(FIND_CONFIG_FIELD, "bar", REPLACE_CONFIG_FIELD, "baz");
+    private static final Integer TIMEOUT_SECONDS = 10;
+    private static final String TOPIC_NAME = "test";
+    private static final Integer TOPIC_PARTITIONS = 1;
+    private static final Short TOPIC_REPLICATION = 1;
+
+    KafkaCluster cluster;
```

Comment:
I thought we needed an annotation to get this injected by the extension?

Reply:
By the looks, if there's no annotation it calculates an empty list of constraints [here](https://github.com/kroxylicious/kroxylicious-junit5-extension/blob/6ec0bb1fcab937d7a97582572dbccfa0a8f8c64f/junit5-extension/src/main/java/io/kroxylicious/testing/kafka/junit5ext/KafkaClusterExtension.java#L973)